1.10/tutorials/dcos-101/service-discovery.md

```diff
@@ -44,31 +62,34 @@ In this section, you will learn about DC/OS service discovery by exploring the d
   redis-1y1hj-s1.marathon.mesos. 60 IN  A 10.0.0.43
   ```
 
-  So you now know that your redis app is running on `10.0.0.43:30585`.
+  So you now know that your Redis app is running on `10.0.0.43:30585`
 
-  <a name="namedvips"></a>
-  * [Named Vips](/docs/1.10/networking/load-balancing-vips/) allow you to assign name/port pairs to your apps. Named VIPs allow you to assign meaningful names to your apps.
-  For example, you can assign a named VIP to your redis service by adding the following to the package definition:
+# Named Virtual IPs
+  
+  * [Named VIPs](/docs/1.10/networking/load-balancing-vips/) allow you to assign name/port pairs to your apps, which means you can give your apps meaningful names with a predictable port. They also provide in-built load balancing when using multiple instances of an application. 
+  For example, you can assign a named VIP to your Redis service by adding the following to the package definition:
 
   ~~~
   "VIP_0": "/redis:6379"
   ~~~
 
   The full name is then generated using the following schema:
-  *vip-name.scheduler.l4lb.thisdcos.directory:vip-port*. In the example above, you can access the redis service from within the cluster at the following address: redis.marathon.l4lb.thisdcos.directory:6379.
+  *vip-name.scheduler.l4lb.thisdcos.directory:vip-port*. 
+
+  As we can see from the example [application](https://raw.githubusercontent.com/joerg84/dcos-101/master/app1/app1.py), this is the mechanism used by the redis package, and so you can access the Redis service from within the cluster at `redis.marathon.l4lb.thisdcos.directory:6379`.
 
 # Outcome
-You know how to use service discovery to connect to your application from within your DC/OS cluster.
+You know how to use service discovery to connect to your application from within your DC/OS cluster, and have learnt about the two mechanisms for service discovery available in DC/OS.
 
 # Deep Dive
 What are the differences between [Mesos-DNS](#mesosdns) and [Named VIPs](#namedvips)?
 
 ## Mesos-DNS
-Mesos-DNS is a rather simple solution to finding applications inside the cluster. While DNS is supported by many applications, Mesos-DNS has the following drawbacks::
+Mesos-DNS is a simple DNS based solution to finding applications inside the cluster. While DNS is supported by many applications, Mesos-DNS has the following drawbacks:
 
-  * DNS caching: Applications sometimes cache DNS entries for efficiency and therefore might not have updated address information (e.g., after a task failure).
-  * You need to use SRV DNS records to retrieve information about the allocated ports. Even though applications commonly understand DNS A records, not all applications support SRV records.
+  * DNS caching: Applications sometimes cache DNS entries for efficiency and therefore might not have updated address information (e.g. after a task failure).
```

Comment:
I had to look this up - I did wonder initially why there were commas after all the e.g.'s and had no idea that this was a US English thing ;) https://jakubmarian.com/comma-after-i-e-and-e-g/ - is comma the standard ? If so I've taken out other ones ....